woocommerce/woocommerce-ios
# Avoid production use of a SwiftUIPreview Sample helper method, which caused a build error when making release builds

Closes: #5708


### Description
We were unintentionally using a helper method intended for use creating Products in `SwiftUI_Previews` methods, and unintentionally using it in production for ghost cells. We were also using it in tests unrelated to SwiftUI.

This fixes the production use by using the standard `Product()` initialiser for the ghost cells, and the test use by adding a `.create()` method to the test target. 

The original SwiftUI helper is also renamed to be specific, so we don't unknowingly use it again.

### Testing instructions

1. Go to Edit Scheme in Xcode and set the Run Build Configuration to Release.
2. Run the app
3. Observe the app builds without an error.


---
